src/components/WwwFrame/TheBasketBar.vue

```diff
@@ -10,7 +10,7 @@
 	>
 		<!--eslint-enable max-len -->
 		<kv-button
-			:to="basketLink"
+			@click="$emit('show-basket')"
```

Comment:
You could change the basket button to be a js driven click by creating a new method that you call here. In the new method, you would handle emitting the event you need and blocking navigation if this.corporate is true.

You would still need to keep the `to` property in place and set to `/basket`.